File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {

        int n = nums.length;
        int counter = 1;
        int candidate = nums[0];
        for(int i = 1; i < n; i++) {
            if(counter == 0) {
                candidate = nums[i];
                counter++;
            }
            else {
                if (candidate == nums[i]) {
                    counter++;
                }
                else {
                    counter--;
                }
            }
        }

        return candidate;
        
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        counter = 0
        currentNum = 0

        for num in nums:
            if counter == 0:
                currentNum = num
            if currentNum == num:
                counter += 1
            else:
                counter -= 1
        return currentNum